src/execution/physical_plan/plan_delim_join.cpp

```diff
@@ -40,7 +50,40 @@ unique_ptr<PhysicalOperator> PhysicalPlanGenerator::PlanDelimJoin(LogicalCompari
 		distinct_groups.push_back(make_uniq<BoundReferenceExpression>(bound_ref.return_type, bound_ref.index));
 	}
 	// now create the duplicate eliminated join
-	auto delim_join = make_uniq<PhysicalDelimJoin>(op.types, std::move(plan), delim_scans, op.estimated_cardinality);
+	auto delim_join =
+	    make_uniq<PhysicalLeftDelimJoin>(op.types, std::move(plan), delim_scans, op.estimated_cardinality);
+	// we still have to create the DISTINCT clause that is used to generate the duplicate eliminated chunk
+	delim_join->distinct = make_uniq<PhysicalHashAggregate>(context, delim_types, std::move(distinct_expressions),
+	                                                        std::move(distinct_groups), op.estimated_cardinality);
+	return std::move(delim_join);
+}
+
+unique_ptr<PhysicalOperator> PhysicalPlanGenerator::PlanRightDelimJoin(LogicalComparisonJoin &op) {
+	// first create the underlying join
+	auto plan = PlanComparisonJoin(op);
```

Comment:
Maybe this can be unified further as this looks like a lot of the same code between left/right with only the index differing?